pkg/apis/pipeline/v1beta1/run_interface.go

```diff
@@ -38,6 +38,11 @@ type RunObject interface {
 	IsCancelled() bool
 	HasStarted() bool
 	IsDone() bool
+}
+
+// RunObject is implemented by Run and CustomRun
+type RunObjectWithRetries interface {
+	RunObject
 
 	GetRetryCount() int
```

Comment:
`PipelineRuns` do not have retries, so splitting `GetRetryCount` out to a separate interface